src/mmap_storage.cpp

```diff
@@ -903,8 +903,17 @@ error_code translate_error(std::system_error const& err, bool const write)
 #endif
 
 		boost::optional<aux::file_view> h = open_file_impl(sett, file, mode, ec);
-		if ((mode & aux::open_mode::write)
-			&& ec.ec == boost::system::errc::no_such_file_or_directory)
+		
+		bool should_retry = (mode & aux::open_mode::write)
+			&& ec.ec == boost::system::errc::no_such_file_or_directory;
+#ifdef _WIN32
+		// this is a workaround for improper handling of files on windows shared drives.
+		// if the directory on a shared drive does not exist, 
+		// windows returns ERROR_IO_DEVICE instead of ERROR_FILE_NOT_FOUND
+		should_retry |= (ec.ec.value() == ERROR_IO_DEVICE);
```

Comment:
this may set `should_retry` even in the case where the `write` flag is not set in `mode`, which doesn't seem correct.
Additionally, it's not safe to *just* check the error code, you need to match the error category as well. Something like this:

```
ec == error_code(ERROR_IO_DEVICE, system_category())
```

If you think it makes sense to break out the condition into a bool rather than placing it directly in the if-condition, that bool should be made `const` as well.

```
bool const should_retry = ... ;
```